extensions/amazon-lambda/deployment/src/main/java/io/quarkus/amazon/lambda/deployment/AmazonLambdaProcessor.java

```diff
@@ -33,16 +39,28 @@
 
     private static final DotName REQUEST_HANDLER = DotName.createSimple(RequestHandler.class.getName());
 
+    private static final DotName NAMED = DotName.createSimple(Named.class.getName());
+
     @BuildStep(applicationArchiveMarkers = { AWS_LAMBDA_EVENTS_ARCHIVE_MARKERS })
-    List<AmazonLambdaClassNameBuildItem> discover(CombinedIndexBuildItem combinedIndexBuildItem,
-            BuildProducer<ReflectiveHierarchyBuildItem> reflectiveClasses) {
-        List<AmazonLambdaClassNameBuildItem> ret = new ArrayList<>();
+    List<AmazonLambdaBuildItem> discover(CombinedIndexBuildItem combinedIndexBuildItem,
+            BuildProducer<AdditionalBeanBuildItem> additionalBeanBuildItemBuildProducer,
+            BuildProducer<ReflectiveHierarchyBuildItem> reflectiveHierarchy,
+            BuildProducer<ReflectiveClassBuildItem> reflectiveClassBuildItemBuildProducer) {
+        AdditionalBeanBuildItem.Builder builder = AdditionalBeanBuildItem.builder().setUnremovable();
+        List<AmazonLambdaBuildItem> ret = new ArrayList<>();
 
         for (ClassInfo info : combinedIndexBuildItem.getIndex().getAllKnownImplementors(REQUEST_HANDLER)) {
             final DotName name = info.name();
+            builder.addBeanClass(name.toString());
+            String cdiName = null;
+            List<AnnotationInstance> named = info.annotations().get(NAMED);
+            if (named != null && !named.isEmpty()) {
+                cdiName = named.get(0).value().asString();
+            }
 
             final String lambda = name.toString();
-            ret.add(new AmazonLambdaClassNameBuildItem(lambda));
+            ret.add(new AmazonLambdaBuildItem(lambda, cdiName));
```

Comment:
what does naming the handler give us exactly?  this feels like a new requirement.

Reply:
i think i figured it out below.  :)